### PR TITLE
Add an audit option to verify is_sorted (off by default)

### PR DESCRIPTION
**Pull request by vedgy**
This option has been discussed in #36.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -127,6 +127,8 @@ conan install timsort/2.0.2
 
 A few configuration macros allow `gfx::timsort` and `gfx::timmerge` to emit diagnostic, which might be helpful to diagnose issues:
 * Defining `GFX_TIMSORT_ENABLE_ASSERT` inserts assertions in key locations in the algorithm to avoid logic errors.
+* Defining `GFX_TIMSORT_ENABLE_AUDIT` inserts assertions that verify pre- and postconditions. This verification can slow the
```
Comment (Morwenn) [inline README.md:130]: I'd go with "These verifications" but that's mostly nitpicking.

**Comment by Morwenn** [comment]
Moving the comparison function here and there is mostly a micro-optimization, I would rather get rid of them where it allows to avoid using `GFX_TIMSORT_MOVE_UNLESS_AUDIT`. I don't think it's worth the reduced readability.

**Comment by vedgy** [comment]
You are right. Fixed and force-pushed.

**Comment by Morwenn** [comment]
As a final tweak I'd probably move the conditional `#include <cassert>` just above the main `GFX_TIMSORT_ENABLE_ASSERT` handling as follows:

```cpp
#if defined(GFX_TIMSORT_ENABLE_ASSERT) || defined(GFX_TIMSORT_ENABLE_AUDIT)
#   include <cassert>
#endif
```

It feels simpler than having to recheck whether `GFX_TIMSORT_ENABLE_ASSERT` was defined a second time.

Otherwise it looks good to me, thanks for the addition 🙂

**Comment by vedgy** [comment]
Indeed. Force-pushed again.

**Comment by Morwenn** [comment]
It's all merged and good.

Did you have other things you wanted to change, or can I start backporting the changes to the branch 1.x.y and release new versions?

**Comment by vedgy** [comment]
No, I don't plan to change anything else. Backporting the new feature to 1.x.y doesn't seem to be a worthwhile undertaking. It is highly unlikely that projects that can't even use C++11 features would want to use `timmerge` all of a sudden.

**Pull request merged**